src/data-display/board/PBoard.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+    <div class="p-board" :class="{ [`${styleType}-style`]: true }">
+        <template v-for="(board, index) in boardList">
+            <p-board-item :key="`board-${index}`"
+                          class="p-board-item"
+                          :class="{ 'first-list-item': index === 0, 'last-list-item': index === boardList.length - 1}"
+                          :left-icon="board.leftIcon"
+                          :icon-button-sets="board.iconButtonSets"
+                          :rounded="board.rounded"
+                          @click="handleClickBoardItem(board, index)"
+            >
+                <template #left-content>
+                    <slot name="item-left-content" v-bind="{...$props, board, index}" />
+                </template>
+                <template #content>
+                    <slot name="item-content" v-bind="{...$props, board, index}" />
+                </template>
+                <template #custom-right-content>
+                    <slot name="item-custom-right-content" v-bind="{...$props, board, index}" />
+                </template>
+            </p-board-item>
+        </template>
+    </div>
+</template>
+
+<script lang="ts">
+import {
+    computed, defineComponent, reactive, toRefs,
+} from 'vue';
+import type { PropType, SetupContext } from 'vue';
+
+import PBoardItem from '@/data-display/board-item/PBoardItem.vue';
+import { BOARD_STYLE_TYPE } from '@/data-display/board/type';
+import type { BoardProps, BoardSet } from '@/data-display/board/type';
+
+
+export default defineComponent<BoardProps>({
+    name: 'PBoard',
+    components: { PBoardItem },
+    props: {
+        styleType: {
+            type: String,
+            default: BOARD_STYLE_TYPE.list,
+            validator(styleType: any) {
+                return Object.values(BOARD_STYLE_TYPE).includes(styleType);
+            },
+        },
+        boardSets: {
+            type: Array as PropType<BoardSet[]>,
+            default: () => [],
+        },
+        pageLimit: {
+            type: Number,
+            default: 10,
+        },
+    },
+    setup(props, { emit }: SetupContext) {
+        const state = reactive({
+            boardList: computed<BoardSet[]>(() => props.boardSets),
+        });
+
+        const handleClickBoardItem = (item: BoardSet, index) => {
+            emit('click', item, index);
```

Comment:
I think it will overlap with the existing click event!

Reply:
You are right. Edited to reflect your comments.